### ISSUE TITLE
Hide Current Location privacy setting

### ISSUE THREAD

**Issue by Zyro1331** [Yellow-Dog-Man/Resonite-Issues#2578]
### Is your feature request related to a problem? Please describe.

I have a large contacts list, and filtering out the people who are "just contacts" versus actual friends of mine is a tough task. 

Imagine you are a relatively popular user, and want to prevent a bunch of contacts from immediately joining when you appear in a session. Or you're a mentor, who just taught a bunch of users previously but don't have the energy to deal with them joining you the next day.

I'd like to propose a privacy feature that allows you to openly hang out with your friends, but without random people from your contacts list interrupting. 

### Describe the solution you'd like

An option in the Privacy settings that allows you to hide your current location. Much like setting your status to Invisible, but without actually appearing offline. Allowing you to receive messages and invites, but require users to ask for an invite before they're allowed to join the session you're currently in. _(A lot like an alternative platform's "Ask Me" status)_

### Describe alternatives you've considered

This feature could pair nicely with the Busy status, making it behave more like the Invisible status. But that might come off more as wanting to be unprovoked and make people less likely to ask for an invite, which would be against the original intent for this feature.

### Additional Context

Related issues:
* #255 
* #1121

### Requesters

@zyro1331

**Comment by lxw404** [comment]
This might be covered directly by: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/915 where Froox mentions:
> We plan to add ability to add one or more tags to each of the contacts. For example "Friends", "Close Friends", "Coworkers" and so on. Each contact can have as many or as little as you want.
> 
> With this tagging in place, we will be able to implement a number of new functions:
> 
> Filtering contacts list by contact tag for easier organization (most obvious one)
> Opening session to contacts with only specific tag (e.g. you open session to just close friends if you don't want to hang out with a wider group)
> Changing online status selectively to specific contact groups - e.g. you show as Online only to close friends, but invisble/busy to everyone else
> And more!

**Comment by shiftyscales** [comment]
Yeah, I think this issue's needs would probably be better covered by the contacts UI rework #1116, and additional features like #1121 as you highlighted, alongside being able to have tags / contact groups you can apply those permissions to as well as on a per-user basis as is covered by #915.
